// nedb-persist/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = nedb;

var _nedb = require('nedb');

var _nedb2 = _interopRequireDefault(_nedb);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function nedb(db) {
  return {
    getItem: function getItem(key, cb) {
      db.findOne({ key: key }, function (err, doc) {
        if (err) {
          cb(err);
        } else {
          cb(null, doc && doc.value);
        }
      });
    },
    setItem: function setItem(key, value, cb) {
      db.update({ key: key }, { key: key, value: value }, { upsert: true }, function (err) {
        return cb(err);
      });
    },
    removeItem: function removeItem(key, cb) {
      db.remove({ key: key }, function (err) {
        return cb(err);
      });
    },
    getAllKeys: function getAllKeys(cb) {
      db.find({}, function (err, docs) {
        if (err) {
          cb(err);
        } else {
          cb(null, docs.map(function (doc) {
            return doc.key;
          }));
        }
      });
    }
  };
}
